### DISCUSSION TITLE
how to use ApiKey to login with web?

### DISCUSSION THREAD

**Discussion by lijiawei0518** [stashapp/stash#3744] [Q&A] [answered]
can i use the URL addition to achieve password free login?

**Comment by Flashy78** [comment]
I don't think you can. It's meant as a key to pass in when making API calls, not for logging into the web UI.

**Comment by dingdongsolong4** [comment, marked as answer]
Stash will allow any request if it has either the `ApiKey` HTTP header or `apikey` URL query parameter set correctly, irrespective of the login status.

For the UI, the query parameter (that is, adding `?apikey=<apikey>` to the URL) will not work, because it would only be sent for the main page load and not for the many additional API requests that are made to retrieve actual page content.

Setting the HTTP header would work, but adding HTTP headers is difficult to do in a browser, although there might be an extension that could achieve that.

If you just want password-free login without removing the password entirely, a much simpler way would just be to set `Security > Maximum Session Age` in the settings to a very large number and then you would only need to log in once.
